include/zephyr/sd/mmc.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for MMC memory card subsystem
+ */
+
+#ifndef ZEPHYR_INCLUDE_SD_MMC_H_
+#define ZEPHYR_INCLUDE_SD_MMC_H_
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sdhc.h>
+#include <zephyr/sd/sd.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Write blocks to MMC card from buffer
+ *
+ * Writes blocks from MMC buffer to MMC card. For best performance, this buffer
+ * should be aligned to CONFIG_SDHC_BUFFER_ALIGNMENT
+ * @param card MMC card to write from
+ * @param wbuf write buffer
+ * @param start_block first block to write to
+ * @param num_blocks number of blocks to write
+ * @retval 0 write succeeded
+ * @retval -EBUSY: card is busy with another request
+ * @retval -ETIMEDOUT: card write timed out
+ * @retval -EIO: I/O error
+ */
+int mmc_write_blocks(struct sd_card *card, const uint8_t *wbuf,
+	uint32_t start_block, uint32_t num_blocks);
+
+/**
+ * @brief Read block from MMC card to buffer
+ *
+ * Reads blocks into MMC buffer from MMC card. For best performance, this buffer
+ * should be aligned to CONFIG_SDHC_BUFFER_ALIGNMENT
+ * @param card MMC card to read from
+ * @param rbuf read buffer
+ * @param start_block first block to read from
+ * @param num_blocks number of blocks to read
+ * @retval 0 read succeeded
+ * @retval -EBUSY: card is busy with another request
+ * @retval -ETIMEDOUT: card read timed out
+ * @retval -EIO: I/O error
+ */
+int mmc_read_blocks(struct sd_card *card, uint8_t *rbuf,
+	uint32_t start_block, uint32_t num_blocks);
+
+/**
+ * @brief Get I/O control data from MMC card
+ *
+ * Sends I/O control commands to MMC card.
+ * @param card MMC card
+ * @param cmd I/O control command
+ * @param buf I/O control buf
+ * @retval 0 IOCTL command succeeded
+ * @retval -ENOTSUP: IOCTL command not supported
+ * @retval -EIO: I/O failure
+ */
+int mmc_ioctl(struct sd_card *card, uint8_t cmd, void *buf);
```

Comment:
what are the set of 'cmd'?  This seems like it should possibly be a enum or at least have #defines in this file.

Reply:
They map to the disk subsystem ioctl calls. I just matched the doc comment to be the same as what sdmmc has for consistency.

Reply:
Is there a reference to the disk subsystem docs for the `cmd` values?

Reply:
`cmd` values are documented [here in the disk subsystem](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/disk.h). Do you think something like:
```
* @param cmd I/O control command.
*     Mirrors disk subsystem, see disk.h for list of possible commands.
```
would be sufficient here?

Reply:
yes, that would be good.  Just something that lets some reading this to know where to look.  I'm sure there some cool way to make an actual reference so generated docs have a link, but that is beyond me.